File: webclientswapi/src/main/java/com/webclient/swapi/webclientswapi/response/dto/Film.java
package com.webclient.swapi.webclientswapi.response.dto;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import java.util.List;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class Film {
    private String title;
    private String episode_id;
    private String opening_crawl;
    private String director;
    private String producer;
    private String release_date;
    private List<String> characters;
    private String url;
}

File: angular-swapi/src/app/interfaces/film.ts
import { People } from './people';

export interface Film {
  results: Result[];
}

export interface Result {
  title: string;
  episode_id: string;
  opening_crawl: string;
  director: string;
  producer: string;
  release_date: string;
  characters: People[];
  url: string;
}
